src/pages/320x240x16/rtc_config.c

```diff
@@ -262,32 +261,32 @@ static const char *rtc_val_cb(guiObject_t *obj, int dir, void *data)
     (void)obj;
     unsigned int idx = (long)data;
     u8 changed;
-    if (idx <= (sizeof(Rtc.value) / sizeof(Rtc.value[0]))) {
-        idx = order[idx];
-        Rtc.value[idx] = GUI_TextSelectHelper(Rtc.value[idx], min[idx], max[idx], dir, 1, 4, &changed);
+    if (idx <= (sizeof(rp->value) / sizeof(rp->value[0]))) {
+        idx = rp->order[idx];
+        rp->value[idx] = GUI_TextSelectHelper(rp->value[idx], rp->min[idx], rp->max[idx], dir, 1, 4, &changed);
         if (changed) {
             if (idx == MONTH || idx == YEAR)
-                max[DAY] = daysInMonth[Rtc.value[MONTH] - 1] + (((Rtc.value[YEAR] % 4) == 0) && (Rtc.value[MONTH] == 2) ? 1 : 0);
+                rp->max[DAY] = daysInMonth[rp->value[MONTH] - 1] + (((rp->value[YEAR] % 4) == 0) && (rp->value[MONTH] == 2) ? 1 : 0);
             GUI_Redraw(&gui->newtime);
             GUI_Redraw(&gui->newdate);
         }
         if (idx == YEAR)
-            sprintf(tempstring, "%4d", Rtc.value[idx]);
+            snprintf(tempstring, 5, "%4d", rp->value[idx]);
         else if (idx == HOUR) {
-            u8 tmp = Rtc.value[HOUR];
+            u8 tmp = rp->value[HOUR];
             if (Transmitter.rtc_timeformat & CLOCK12HR) {
                 tmp %= 12;
                 if (tmp == 0)
                     tmp = 12;
-                sprintf(tempstring, "%2d %s", tmp, Rtc.value[HOUR] >= 12? "pm" : "am");
+                snprintf(tempstring, 6, "%2d %s", tmp, rp->value[HOUR] >= 12? "pm" : "am");
             } else {
-                sprintf(tempstring, "%2d", tmp);
+                snprintf(tempstring, 3, "%2d", tmp);
```

Comment:
Lint Error:
```
281:  If you can, use sizeof(tempstring) instead of 6 as the 2nd arg to snprintf.  
283:  If you can, use sizeof(tempstring) instead of 3 as the 2nd arg to snprintf.  
```